ext/pdo_pgsql/pgsql_driver.c

```diff
@@ -102,9 +102,16 @@ int _pdo_pgsql_error(pdo_dbh_t *dbh, pdo_stmt_t *stmt, int errcode, const char *
 }
 /* }}} */
 
-static void _pdo_pgsql_notice(pdo_dbh_t *dbh, const char *message) /* {{{ */
+static void _pdo_pgsql_notice(void *context, const char *message) /* {{{ */
 {
-/*	pdo_pgsql_db_handle *H = (pdo_pgsql_db_handle *)dbh->driver_data; */
+	zval zarg;
+	pdo_dbh_t * dbh = (pdo_dbh_t *)context;
+	zend_fcall_info_cache *fc = ((pdo_pgsql_db_handle *)dbh->driver_data)->notice_callback;
+	if (fc) {
+		ZVAL_STRING(&zarg, message);
+		zend_call_known_fcc(fc, NULL, 1, &zarg, NULL);
+		zval_ptr_dtor(&zarg);
```

Comment:
Come to think of it, you _could_ use `zval_ptr_dtor_str` here, but it doesn't really matter I guess.